### ISSUE TITLE
Aperture photometry tables act like scalar apertures with length-1 list

### ISSUE THREAD

**Issue by eteq** [astropy/photutils#705]
Some behavior pointed out to me by @hcferguson that I think is counter-intuitive and should change.  First, the part that *does* make sense:

```python
#setup
>>> im = np.random.randn(100,100)
>>> ap = photutils.CircularAperture((50,50),5)

>>> photutils.aperture_photometry(im, ap).colnames
['id', 'xcenter', 'ycenter', 'aperture_sum']
>>> photutils.aperture_photometry(im, [ap, ap]).colnames
['id', 'xcenter', 'ycenter', 'aperture_sum_0', 'aperture_sum_1']
```
So far so good. But then the following:
```python
>>> photutils.aperture_photometry(im, [ap]).colnames
['id', 'xcenter', 'ycenter', 'aperture_sum']
```
`aperture_sum` does not have an ``_0``!  This is very counter-intuitive, as normally a distinction is drawn between "scalar" and "list/array/whatever of length 1". 

I'd classify this as a bug and can just be fixed - do you agree @larrybradley ?  It does have potential backwards-compatibility concerns, but still... This is a thing that trips up enough novice users (the length-1 vs scalar distinction) that I think it's worth it.

**Comment by larrybradley** [comment]
Yes, I agree that we should fix that.

**Comment by larrybradley** [comment]
Fixed in #779.
